paddle/fluid/distributed/auto_parallel/spmd_rules/split_spmd_rule.cc

```diff
@@ -0,0 +1,126 @@
+/* Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/fluid/distributed/auto_parallel/spmd_rules/split_spmd_rule.h"
+#include <algorithm>
+#include <typeinfo>
+#include "paddle/phi/core/distributed/auto_parallel/utils.h"
+
+namespace paddle {
+namespace distributed {
+namespace auto_parallel {
+
+using phi::distributed::auto_parallel::str_join;
+
+std::pair<std::vector<TensorDistAttr>, std::vector<TensorDistAttr>>
+SplitSPMDRule::InferForward(const std::vector<DistTensorSpec>& input_specs,
+                            const paddle::framework::AttributeMap& attrs) {
+  // step0: Verify Input Args Based on Elementwise Logic
+  int64_t ninputs = input_specs.size();
+  PADDLE_ENFORCE_EQ(
+      ninputs,
+      1,
+      phi::errors::InvalidArgument("The size of InputSpec in split must "
+                                   "be equal to 1, but got [%d].",
+                                   ninputs));
+  VerifySpecs(input_specs, "split");
+
+  // step1: Build Einsum Notation
+  int64_t ndim = input_specs[0].shape().size();
+  Attribute section_attr = GetAttr("num_or_sections", attrs);
+  int64_t noutput = 0;
+  if (section_attr.type() == typeid(int)) {
+    noutput = *paddle::framework::ExtractAttribute<int>("num_or_sections")(
+        section_attr);
+  } else if (section_attr.type() == typeid(std::vector<int>)) {
+    std::vector<int64_t>* sections =
+        paddle::framework::ExtractAttribute<std::vector<int64_t>>(
+            "num_or_sections")(section_attr);
+    noutput = sections->size();
+  }
+  int64_t axis = ExtractAttr<int>("axis", attrs);
+  if (axis < 0) {
+    axis += ndim;
+  }
+  std::string alphabet = "abcdefghijklmnopqrstuvwxyz";
+
+  // get einsum notation for input
+  std::vector<std::string> input_axes_vec;
+  std::string input_axes = alphabet.substr(0, ndim);
+  input_axes_vec.emplace_back(input_axes);
+
+  // get einsum notation for output
+  std::string output_axes(input_axes);
```

Comment:
we should support input tensor sharded on split_axis.
the main logic for split spmd forward infer should be:

1. for axes other than the split_axis, treat as broadcast axes, copy dims from input to  outputs.
2. for split_axis:  
     the infered dim_mapping for all outputs are replicated
     the infered dst_dim_mapping for all outputs is also replicated

Reply:
use a special char for split axis, like: "k"

Reply:
Done